### PR TITLE
Enable new exception handling by default

### PR DESCRIPTION
**Pull request by janvorli**
This change just flips the config knob to enable the new exception handling by default.

### CODE REVIEW

File: src/coreclr/inc/clrconfigvalues.h [hunk 1]
```diff
@@ -259,7 +259,7 @@ RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_legacyCorruptedStateExceptionsPolicy, W("le
 CONFIG_DWORD_INFO(INTERNAL_SuppressLostExceptionTypeAssert, W("SuppressLostExceptionTypeAssert"), 0, "")
 RETAIL_CONFIG_DWORD_INFO(INTERNAL_UseEntryPointFilter, W("UseEntryPointFilter"), 0, "")
 RETAIL_CONFIG_DWORD_INFO(INTERNAL_Corhost_Swallow_Uncaught_Exceptions, W("Corhost_Swallow_Uncaught_Exceptions"), 0, "")
-RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableNewExceptionHandling, W("EnableNewExceptionHandling"), 0, "Enable new exception handling.");
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableNewExceptionHandling, W("EnableNewExceptionHandling"), 1, "Enable new exception handling.");
```
Comment (mangod9) [inline src/coreclr/inc/clrconfigvalues.h:262]: is there a way to configure this via runtimeconfig as well? We had some asks for having even W^X configurable through that as well.

Comment (janvorli) [reply src/coreclr/inc/clrconfigvalues.h:262]: No, let me figure out how to do that.

Comment (janvorli) [reply src/coreclr/inc/clrconfigvalues.h:262]: @mangod9  I've just added a commit that adds "System.Runtime.EnableNewExceptionHandling" setting that can control that from runtime.config.

**Comment by mangod9** [comment]
Looks like there are PALTest failures?

File: src/coreclr/inc/clrconfigvalues.h (hunk 1 above)
Comment (davidwrighton) [reply src/coreclr/inc/clrconfigvalues.h:262]: Could we not call this "New" exception handling? I've worked on way too many features where the "New" thing was 15 years old and its name confused everyone at all times.

Comment (davidwrighton) [reply src/coreclr/inc/clrconfigvalues.h:262]: Ah, I suppose thats the name throughout the product at this point. Oh well. I suppose New it is. Sigh.

Comment (janvorli) [reply src/coreclr/inc/clrconfigvalues.h:262]: @davidwrighton yeah, I know. I couldn't come up with anything better over the time. If you have a better idea, I'd be happy to rename it.

Comment (janvorli) [reply src/coreclr/inc/clrconfigvalues.h:262]: Just a quick idea - would using `DOTNET_ExceptionHandlingMode` and `System.Runtime.ExceptionHandlingMode` with a numeric value feel better? We would use 0 for the old EH, 1 for the new EH and if we ever moved it to something else, then we could keep incrementing it.

**Comment by janvorli** [comment]
> Looks like there are PALTest failures?

Yes, break from another PAL refactoring, see #98574

**Comment by janvorli** [comment]
I don't know what happened. Huge number of tests are failing. I had a test PR before with all the changes and all tests were passing. I must have merged something incorrectly. I need to investigate it.

File: src/coreclr/inc/clrconfigvalues.h (hunk 1 above)
Comment (jkotas) [reply src/coreclr/inc/clrconfigvalues.h:262]: > I couldn't come up with anything better over the time. 

A better name would be `LegacyExceptionHandling` with opposite meaning. It makes it clear that it is something that one should not use and that is likely going to disappear at some point.

**Review by mangod9 — APPROVED** [review]

**Pull request merged**